src/handlers/int-test-support/helpers/s3Helper.ts

```diff
@@ -93,45 +118,82 @@ const putS3ObjectBasic = async (
 
 const putS3Object = callWithRetryAndTimeout(putS3ObjectBasic);
 
-const deleteS3ObjectBasic = async (
-  object: S3Object
-): Promise<DeleteObjectCommandOutput> => {
+const deleteS3ObjectsBasic = async (
+  params: deleteS3ObjectByPrefixes
+): Promise<DeleteObjectsCommandOutput> => {
   if (runViaLambda())
     return (await sendLambdaCommand(
-      IntTestHelpers.deleteS3Object,
-      object
-    )) as unknown as DeleteObjectCommandOutput;
-
-  const bucketParams = {
-    Bucket: object.bucket,
-    Key: object.key,
+      IntTestHelpers.deleteS3Objects,
+      params
+    )) as unknown as DeleteObjectsCommandOutput;
+  const result: DeleteObjectsCommandOutput = {
+    Deleted: [],
+    Errors: [],
+    $metadata: {},
   };
-  return await s3Client.send(new DeleteObjectCommand(bucketParams));
+  for (const prefixToDelete of params.prefixesToDelete) {
+    const listResult = await listS3Objects({
+      bucketName: params.bucketName,
+      prefix: prefixToDelete,
+    });
+    if (listResult.Contents && listResult.Contents.length > 0) {
+      const keysToDelete = listResult.Contents.map(({ Key }) => ({ Key }));
+      const deleteResult = await deleteS3Object({
```

Comment:
This will probably fail now if we have a large number of objects with a single prefix, but let's worry about that when we need it. The "list 1000, then delete 1000" approach was actually better and we could have made it work through the lambda, but I guess that's a bit much for now.